src/LocalNeuralNet.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef LOCALNEURALNET_HPP_INCLUDE
+#define LOCALNEURALNET_HPP_INCLUDE
+
+#include <memory>
+#include <vector>
+
+namespace geopm
+{
+    ///  @brief Class to manage data and operations of feed forward neural nets
+    ///         required for neural net inference.
+
+    class LocalNeuralNet
+    {
+        public:
+            /// @brief Returns a unique_ptr to a concrete object
+            ///        constructed using the underlying implementation
+            ///        from an array of doubles of weights and biases.
+            /// 
+            /// @param [in] vector of pairs of weight matrices (as
+            ///             double vectors of floats) and bias vectors
+            ///             as vectors of floats.
+            /// 
+            /// @throws geopm::Exception if dimensions are incompatible.
+            static std::unique_ptr<LocalNeuralNet> make_unique(std::vector<std::pair<std::vector<std::vector<float> >, std::vector<float> > > input);
```

Comment:
Suggestion: given the complexity of the parameter's type, for readability, use typedefs to simplify its declaration and at the same time describe what it is. For example:

`typedef std::pair<...> Layer // or NNLayer, NeuralNetLayer... (I'm sure you can think of a nicer name)`

or

```
typedef std::vector<std::vector<float> > Weights
typedef std::vector<float> Biases
typedef std::pair<Weights, Biases> Layer
```

This way the method declaration would naturally be more readable:

`static std::unique_ptr<LocalNeuralNet> make_unique(std::vector<Layer> input)`